File: src/main/java/workshop/part2/subactor/VettingActorFactory.java
package workshop.part2.subactor;

import java.util.concurrent.TimeUnit;

import akka.actor.ActorContext;
import akka.actor.ActorRef;
import akka.actor.Props;
import scala.concurrent.duration.Duration;

public class VettingActorFactory {

    private ActorRef userActor;
    private ActorRef fraudWordActor;

    public VettingActorFactory(ActorRef userActor, ActorRef fraudWordActor) {
        this.userActor = userActor;
        this.fraudWordActor = fraudWordActor;
    }

    public ActorRef create(ActorContext context) {
        return context.system().actorOf(Props.create(VettingActor.class,
            () -> new VettingActor(userActor, fraudWordActor, Duration.create(1, TimeUnit.SECONDS))));
    }
}

File: src/main/java/workshop/part2/subactor/VettingActorFactory.kt
package workshop.part2.subactor

import akka.actor.ActorContext
import akka.actor.ActorRef
import akka.actor.Props
import scala.concurrent.duration.Duration
import java.util.concurrent.TimeUnit

open class VettingActorFactory(private val userActor: ActorRef, private val fraudWordActor: ActorRef) {

    open fun create(context: ActorContext): ActorRef {
        return context.system().actorOf(Props.create(VettingActor::class.java
        ) { VettingActor(userActor, fraudWordActor, Duration.create(1, TimeUnit.SECONDS)) })
    }
}